models/model.js

```diff
@@ -50,18 +51,44 @@ async function selectCommentsByArticle(article_id){
     WHERE articles.article_id = $1`,[article_id])
   
     return rows.length >0 ? rows : Promise.reject({status:404, msg: "Comments with this Article Id dont exist"})
-    
-        
-        
-    
-    
-    
+}
+async function insertComment(commentinfo,articleId){
+    try{
+      
+        if((await db.query('SELECT * FROM articles WHERE article_id = $1',[articleId])).rows.length ===0 ){
+            return Promise.reject({status:404, msg: "Article not found"})
+        }
+        if(JSON.stringify(Object.keys(commentinfo)) !== '["username","body"]'){
 
+            return Promise.reject({status:400, msg: "Bad Request: Incorrect format for comment"})
+        }
+        if(typeof commentinfo.username !== 'string' || typeof commentinfo.body !== 'string'){
+            console.log(commentinfo)
+            return Promise.reject({status:400, msg: "Bad Request: Incorrect format for comment"})
+        }
```

Comment:
Nothing really wrong with this approach, but would PSQL not give us an error if we try and input the wrong data type?